verification/deriveChild.go

```diff
@@ -0,0 +1,320 @@
+	if resp, err = c.DeriveChild(handle, make([]byte, digestLen), 0, 0, 0); err != nil {
+		t.Errorf("[ERROR]: Error while creating child context handle: %s", err)
+	}
+	handle = &resp.NewContextHandle
+	if resp.NewContextHandle == DefaultContextHandle {
+		t.Fatalf("[FATAL]: Incorrect handle. Should return non-default handle, but returned %v", *handle)
+	}
+	if resp.ParentContextHandle != InvalidatedContextHandle {
+		t.Errorf("[ERROR]: Incorrect handle. Should be invalidated when retain parent is NOT set, but returned %v", resp.ParentContextHandle)
+	}
+
+	// Now, there is no default handle, setting RetainParent flag should succeed
+	if resp, err = c.DeriveChild(handle, make([]byte, digestLen), RetainParent, 0, 0); err != nil {
+		t.Errorf("[ERROR]: Error while making child context handle as default handle: %s", err)
+	}
+	handle = &resp.NewContextHandle
+	parentHandle := resp.ParentContextHandle
+
+	if parentHandle == InvalidatedContextHandle {
+		t.Errorf("[ERROR]: Incorrect handle. Should return retained handle, but returned invalidated handle %v", parentHandle)
+	}
+}
+
+// Validates DerivedChild command with ChangeLocality flag.
+func TestChangeLocality(d TestDPEInstance, c DPEClient, t *testing.T) {
+	if !d.HasLocalityControl() {
+		t.Skip("WARNING: DPE profile does not have control over locality. Skipping this test...")
+	}
+
+	var resp *DeriveChildResp
+	simulation := false
+	handle := getInitialContextHandle(d, c, t, simulation)
+	// Clean up contexts
+	defer func() {
+		err := c.DestroyContext(handle, DestroyDescendants)
+		if err != nil {
+			t.Errorf("[ERROR]: Error while cleaning contexts, this may cause failure in subsequent tests: %s", err)
+		}
+	}()
+
+	// Get digest size
+	profile, err := GetTransportProfile(d)
+	if err != nil {
+		t.Fatalf("Could not get profile: %v", err)
+	}
+
+	digestLen := profile.GetDigestSize()
+	currentLocality := d.GetLocality()
+	otherLocality := currentLocality + 1
+
+	// Create child context in other locality
+	if resp, err = c.DeriveChild(handle, make([]byte, digestLen), ChangeLocality, 0, otherLocality); err != nil {
+		t.Fatalf("[ERROR]: Error while creating child context handle in other locality: %s", err)
+	}
+	handle = &resp.NewContextHandle
+
+	// Revert to same locality from other locality
+	prevLocality := currentLocality
+	d.SetLocality(otherLocality)
+
+	if resp, err = c.DeriveChild(handle, make([]byte, digestLen), ChangeLocality, 0, prevLocality); err != nil {
+		t.Fatalf("[FATAL]: Error while creating child context handle in previous locality: %s", err)
+	}
+	handle = &resp.NewContextHandle
+	d.SetLocality(prevLocality)
+}
+
+// Checks whether the DeriveChild input flags - InternalDiceInfo, InternalInputInfo are supported
+// while creating child contexts when these features are supported in DPE profile.
+func TestInternalInputFlags(d TestDPEInstance, c DPEClient, t *testing.T) {
+	var resp *DeriveChildResp
```

Comment:
Added a separate test case for Input flags as they may/may not be supported by profile